src/computability/partrec_code.lean

```diff
@@ -482,8 +482,8 @@ end
 def eval : code → ℕ →. ℕ
 | zero         := pure 0
 | succ         := nat.succ
-| left         := λ n, n.unpair.1
-| right        := λ n, n.unpair.2
+| left         := ↑(λ n : ℕ, n.unpair.1)
+| right        := ↑(λ n : ℕ, n.unpair.2)
```

Comment:
There's a very similar timeout issue here: https://github.com/leanprover-community/mathlib/blob/master/src/computability/partrec.lean#L149

Reply:
Do you have an idea why this became an issue after #2363?

Reply:
Unfortunately, no. Have you looked at type class search traces before and after?